### PR TITLE
feat: Make ConfigStore aware of known partitions

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This adds a `update_partitions` method the allows the config store to be told what partitions we are responsible for. This method will automatically drop partitions that we become no longer responsible for.

### CODE REVIEW

**Review by davidenwang — APPROVED** [review]

**Review by davidenwang — COMMENTED** [review]
overall makes sense to me, but is there a reason we have to keep track of known_partitions separately? rather than relying on the keys of partitioned_buckets?

**Comment by evanpurkhiser** [comment]
> overall makes sense to me, but is there a reason we have to keep track of known_partitions separately? rather than relying on the keys of partitioned_buckets?

Good point. That does seem better.

**Review by davidenwang — APPROVED** [review]
nice nice

File: src/config_store.rs [hunk 1]
```diff
@@ -121,8 +134,39 @@ impl ConfigStore {
...
+        for new_partiton in new_partitions.difference(&known_partitions) {
+            self.partitioned_buckets
+                .entry(*new_partiton)
+                .or_insert_with(make_empty_tick_buckets);
+        }
+    }
+
+    /// Drop an entire partition of check configurations. This does NOT update the known set of
+    /// partitions. Use update_partitions.
+    fn drop_partition(&mut self, partition: u16) {
```
Comment (wedamija) [inline src/config_store.rs:169]: Should this be private so that no one external can call it?

Comment (evanpurkhiser) [reply src/config_store.rs:169]: It is, you need `pub` to make it public.

**Pull request merged**